sdks/go/pkg/beam/core/runtime/symbols.go

```diff
@@ -26,12 +26,14 @@ import (
 )
 
 var (
-	Resolver SymbolResolver
-	cache    = make(map[string]interface{})
-	mu       sync.Mutex
+	// Resolver is the accessible symbol resolver the runtime uses to find functions.
+	Resolver         SymbolResolver
```

Comment:
IIRC this is public to allow it to be overwritten. If we we do lazy init, then we would break such attempts. It might be better to add a SymbolResolver that lazily calls symtab.New. Then it works no matter how it is used.

Reply:
Good catch! Thanks for the suggestion!